packages/frontend/src/pages/Space.tsx

```diff
@@ -317,6 +317,11 @@ const Space: FC = () => {
                         icon: <IconLayoutDashboard size={30} />,
                         title: 'No items added yet',
                     }}
+                    defaultActiveTab={
+                        !dashboards || dashboards.length === 0
+                            ? 'charts'
+                            : 'dashboards'
+                    }
```

Comment:
we can actually shift the logic and move it inside resource view to select a tab based on the items passed to it. we'll no longer need to have `defaultActiveTab` manually supplied to the component @yasmin-e thoughts?

Reply:
Hello @IrakliJani, will implement and update here. Please let me know if something different is required

Reply:
@CVamsi27 
yep, I think it will be best to iterate over tabs and also check for items.

for example:

check for charts: 0 items (skip)
check for dashboards: 1 item (set)

make sure if both charts and dashboards are 0 then the empty state shows up

Reply:
cc @IrakliJani @CVamsi27 
🤔  considering ResourceView will get more complex soon, having the prop can be good but I think we should still check inside the comp as default behaviour like @ZeRego suggests.

Reply:
@ZeRego @yasmin-e overriding sounds good. since `ResourceView` is used in a lot of places I would love to have some kind of consistency and a sense that every page which uses `ResourceView` is behaving the same way. in this PR and referenced issue we are only addressing problem on the Spaces page. but I am guessing same problem might also exist on the home page (where we have `ResourceView` with tabs).

I think @ZeRego suggestion is the best of both ideas:

- setting default tab that has first found resource (length > 0)
- having an override `defaultActiveTab` which will take a priority ^

cc: @CVamsi27

Reply:
Hello @IrakliJani, should I change something in this PR or its good to go?

Reply:
hi @CVamsi27 it will be good if you can incorporate some of the changes that we discussed here. we agreed to have  `defaultActiveTab` but it will also be good if ResourceView detects which tab to set as a default if `defaultActiveTab` is not provided to the component. happy to take it from what you've already done or wait for your changes

Reply:
Hello @IrakliJani, I am little confused here and a little busy this weekend. Please go ahead and make the changes. Sorry for the inconvenience.
Thanks, Should I close this PR?